presto-main/src/main/java/io/prestosql/sql/ParameterUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.sql.tree.AstVisitor;
+import io.prestosql.sql.tree.Expression;
+import io.prestosql.sql.tree.Node;
+import io.prestosql.sql.tree.NodeRef;
+import io.prestosql.sql.tree.Parameter;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class ParameterUtils
+{
+    private ParameterUtils() {}
+
+    public static Map<NodeRef<Expression>, Integer> parameterExtractor(Node node)
+    {
+        List<Expression> parameters = ImmutableList.sortedCopyOf(new ExpressionComparator(), new AstVisitor<List<Expression>, Void>()
+        {
+            @Override
```

Comment:
There's already a class that does this: `ParameterExtractor`.

This could all be replaced with:

```java
List<Parameter> parameters = getParameters(...).stream()
        .sorted(Comparator.comparing(
                parameter -> parameter.getLocation().get(),
                Comparator.comparing(NodeLocation::getLineNumber)
                        .thenComparing(NodeLocation::getColumnNumber)))
        .collect(Collectors.toList());
```